modules/quirks/qualcomm/sdm845-modem.nix

```diff
@@ -2,10 +2,41 @@
 
 let
   cfg = config.mobile.quirks.qualcomm;
-  inherit (lib) mkIf mkOption types;
+  inherit (lib)
+    any
+    id
+    mkIf
+    mkOption
+    optional
+    types
+  ;
+  anyCompatible = any id [
+    cfg.sdm845-modem.enable
+    cfg.sc7180-modem.enable
+  ];
+
+  # Systems for which we read the partition directly.
+  # TODO: This is likely the wrong abstraction.
+  # Once we have more accrued knowledge, add a discrete option.
+  rmtfsReadsPartition = any id [
+    cfg.sdm845-modem.enable
+  ];
+
+  # TODO: figure out what PD mapper exactly is...
+  # is it USB PD or something modem related?
+  withPDMapper = any id [
+    cfg.sdm845-modem.enable
+  ];
```

Comment:
Given that I don't know yet what the proper abstraction is, let's keep it this way. Clearly fenced as "dunno".

I guess we'll need more devices that needs the same basic services around to know whether it's an SoC abstraction, or a device abstraction.